src/PhpParser/AstResolver.php

```diff
@@ -105,14 +106,15 @@ public function resolveClassMethodFromMethodReflection(MethodReflection $methodR
             return null;
         }
 
-        $class = $this->betterNodeFinder->findFirstInstanceOf($nodes, Class_::class);
-        if (! $class instanceof Class_) {
+        $classLike = $this->betterNodeFinder->findFirstInstanceOf($nodes, ClassLike::class);
```

Comment:
`findFirstInstanceOf` seems not really a solution if interface and class position are flipped:

```php
class ImplementsInterface implements InterfaceWithParam {
    public function execute()
    {
    }
}

/**
 * Both interface and class on same file on purpose to ensure locate
 * parent interface bug
 */
interface InterfaceWithParam
{
    public function execute($foo);
}
```